crates/nargo_cli/src/cli/compile_cmd.rs

```diff
@@ -125,13 +123,30 @@ pub(crate) fn compile_circuit<B: Backend>(
     compile_options: &CompileOptions,
 ) -> Result<CompiledProgram, CliError<B>> {
     let mut context = resolve_root_manifest(program_dir)?;
-    let result = compile_main(
-        &mut context,
+    let result = compile_main(&mut context, compile_options);
+    let mut program = report_errors(result, &context, compile_options.deny_warnings)?;
+
+    // Apply backend specific optimizations.
+    program.circuit = optimize_circuit(backend, program.circuit).unwrap();
+    Ok(program)
+}
+
+pub(super) fn optimize_circuit<B: Backend>(
+    backend: &B,
+    circuit: Circuit,
+) -> Result<Circuit, CliError<B>> {
+    // Note that this makes the `CircuitSimplifier` a noop.
+    // The `CircuitSimplifier` should be reworked to not rely on values being inserted during ACIR gen.
+    let simplifier = CircuitSimplifier::new(0);
```

Comment:
we should probably just remove this from the interface in a later PR and rework it to be like other optimizations